cpp-ch/local-engine/Parser/SerializedPlanParser.cpp

```diff
@@ -1317,13 +1317,18 @@ void SerializedPlanParser::parseFunctionArguments(
         // In clickhosue, map element are also accessed by arrayElement, not make the cast.
         parseFunctionArgument(actions_dag, parsed_args, required_columns, function_name, args[0]);
         auto element_type = actions_dag->getNodes().back().result_type;
+        const auto * first_arg_type = parsed_args.back()->result_type.get();
         const auto * nested_type = element_type.get();
         if (nested_type->isNullable())
         {
             nested_type = typeid_cast<const DB::DataTypeNullable *>(nested_type)->getNestedType().get();
         }
+        if (first_arg_type->isNullable())
+        {
+            first_arg_type = typeid_cast<const DB::DataTypeNullable *>(first_arg_type)->getNestedType().get();
+        }
         const auto * index_node = parseFunctionArgument(actions_dag, required_columns, function_name, args[1]);
-        if (nested_type->getTypeId() == DB::TypeIndex::Array)
+        if (nested_type->getTypeId() == DB::TypeIndex::Array && first_arg_type->getTypeId() != DB::TypeIndex::Map)
```

Comment:
`nested_type` and `first_arg_type` are both the non-nullable data type of first arg. The second condition is always true if the first conditions is true.

Reply:
fix done.